### PR TITLE
Added Unit tests

### PR DESCRIPTION
**Pull request by jm5967a**
Added some unit tests for index.js. Feel free to suggest any changes that should be made. 

@xiaody

### CODE REVIEW

File: tests/components/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/* eslint-disable no-undef */
+import React from 'react'
+import { shallow } from 'enzyme'
+
+import LinesEllipsis from '../../src/index'
```
Comment (klammm) [inline tests/components/index.spec.js]: NIT: You can leave out the `index` here. 

`import LinesEllipsis from '../../src'`

File: package.json [hunk 2]
```diff
@@ -28,6 +28,8 @@
   "license": "MIT",
   "repository": "xiaody/react-lines-ellipsis",
   "peerDependencies": {
+    "preact": "8.2.9",
```
Comment (ajoshi0) [inline package.json]: undo this change keep them under dev-dependencies

File: src/index.js [hunk 3]
```diff
@@ -105,7 +105,7 @@ class LinesEllipsis extends React.PureComponent {
       clamped,
       text: clamped ? this.units.slice(0, ellipsisIndex).join('') : props.text
     }
-    this.setState(newState, props.onReflow.bind(this, newState))
+    this.setState(newState, props.onReflow && props.onReflow.bind(this, newState))
```
Comment (xiaody) [inline src/index.js]: `defaultProps.onReflow` has been set as a no-op function, so I think the `&&` test is not necessary.

File: tests/components/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
+/* eslint-disable no-undef */
```
Comment (xiaody) [inline tests/components/index.spec.js]: Maybe `/* eslint-env jest */`?  https://eslint.org/docs/user-guide/configuring#specifying-environments

File: src/index.js (hunk 3 above)
Comment (jm5967a) [reply src/index.js]: Nice catch, I added that early on and forgot to remove it. Let me know if there is anything else you would like me to change.

File: tests/components/responsiveHOC.spec.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+/* eslint-disable no-undef */
```
Comment (xiaody) [inline tests/components/responsiveHOC.spec.js]: `eslint-env` too?

Comment (jm5967a) [reply tests/components/responsiveHOC.spec.js]: Fixed

**Comment by xiaody** [comment]
Thanks @jm5967a and everyone!
